### PR TITLE
EL-2624 - Added ability to add custom validation to the integrated da…

### PR DESCRIPTION
**Pull request by AlastairMcKee**
…te picker

Added some documentation to explain the new options, it has not been added to the examples. Everything else mentioned in the ticket is possible as seen in the date range picker documentation.

### CODE REVIEW

File: docs/app/pages/components/sections/date-time-picker/integrated-date-picker-ng1/integrated-date-picker-ng1.component.html [hunk 1]
```diff
@@ -27,6 +27,13 @@
             <td>false</td>
         </tr>
         <tr>
+            <td class="attribute">ng-model-options</td>
```
Comment (ashley-hunter) [inline docs/app/pages/components/sections/date-time-picker/integrated-date-picker-ng1/integrated-date-picker-ng1.component.html]: Remove this description as ng-model-options is a built in Angular feature, not part of this directive.

File: docs/app/pages/components/sections/date-time-picker/integrated-date-picker-ng1/integrated-date-picker-ng1.component.html [hunk 2]
```diff
@@ -84,6 +91,13 @@
                 and <code>end</code>.</td>
             <td>true</td>
         </tr>
+        <tr>
+            <td class="attribute">validator</td>
+            <td>function</td>
+            <td>variable</td>
+            <td>Set a funtion here to override the default validation checks with your own custom checks.</td>
```
Comment (ashley-hunter) [inline docs/app/pages/components/sections/date-time-picker/integrated-date-picker-ng1/integrated-date-picker-ng1.component.html]: I assume this function is expected to return true or false depending on the validity, should maybe mention this.

File: src/ng1/directives/dateTimePicker/dateTimePickerPopup.controller.js [hunk 3]
```diff
@@ -134,7 +134,14 @@ export default function DateTimePickerPopupCtrl($scope, $parse, $attrs) {
             }
         }
 
-        var valid = date.isValid();
+        var valid;
```
Comment (ashley-hunter) [inline src/ng1/directives/dateTimePicker/dateTimePickerPopup.controller.js]: Could simplify this to:

```typescript
var valid = $attrs.validator ? $parse($attrs.validator)($scope)() : date.isValid();
```

**Comment by AlastairMcKee** [comment]
Made requested changes

**Pull request merged**